tests/failpoint/failpoint.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package failpoint
+
+import (
+	gofail "go.etcd.io/gofail/runtime"
+)
+
+type Failpoint interface {
+	Enable() error
+	Disable() error
+}
```

Comment:
It doesn't make much sense to define an interface, because `failpointFunc` is only used in the same package. It makes sense if you move the interface into the test framework. 

We can keep it as it's for now, and discuss it separately.